packages/api/src/domain/medical/__tests__/demographics.test.ts

```diff
@@ -0,0 +1,660 @@
+import { Patient, PatientDemoData, GenderAtBirth, splitDob } from "@metriport/core/domain/patient";
+import { LinkDemographics } from "@metriport/core/domain/patient-demographics";
+import { USState } from "@metriport/core/domain/geographic-locations";
+import {
+  normalizeDob,
+  normalizeGender,
+  normalizeAndStringifyNames,
+  normalizeAddress,
+  stringifyAddress,
+  normalizeTelephone,
+  normalizeSsn,
+  normalizeEmail,
+  normalizeAndStringifyDriversLicense,
+  patientToNormalizedCoreDemographics,
+  createAugmentedPatient,
+  linkHasNewDemographics,
+} from "../patient-demographics";
+
+describe("normalization", () => {
+  it("check dob split", async () => {
+    const dobValidValue = "2023-08-01";
+    const dobSplit = splitDob(dobValidValue);
+    expect(dobSplit).toMatchObject(["2023", "08", "01"]);
+  });
+  it("check dob normalization", async () => {
+    const dobValidValue = "2023-08-01";
+    const dobValid = normalizeDob(dobValidValue);
+    expect(dobValid).toBe(dobValidValue);
+    const dobTrim = normalizeDob(" 2023-08-01 ");
+    expect(dobTrim).toBe(dobValidValue);
+    const dobUndefined = normalizeDob();
+    expect(dobUndefined).toBe(undefined);
```

Comment:
A cursory look in db shows that a bunch of dob results look like this. 
19500503
with no dashes. Can we handle those cases and can we add a test?

Reply:
Done.